### PR TITLE
Batch compatibility for 6 Ethernet elements

### PR DESCRIPTION
**Pull request by gkatsikas**
Signed-off-by: Georgios Katsikas <katsikas.gp@gmail.com>

### CODE REVIEW

**Comment by tbarbette** [comment]
Seems good, thanks !

File: elements/ethernet/arpfaker.cc [hunk 1]
```diff
@@ -57,9 +63,18 @@ ARPFaker::initialize(ErrorHandler *)
...
 {
-    if (Packet *p = ARPResponder::make_response(_eth1.data(), _ip1.data(),
-						_eth2.data(), _ip2.data()))
-	output(0).push(p);
+    Packet *p = ARPResponder::make_response(
+        _eth1.data(), _ip1.data(), _eth2.data(), _ip2.data()
+    );
+
+    if (p) {
+    #if HAVE_BATCH
```
Comment (tbarbette) [inline elements/ethernet/arpfaker.cc]: This is not sufficient, as a path may be not batch-enabled, there is the in_batch_mode() function to check that

File: elements/ethernet/etherpausesource.cc [hunk 2]
```diff
@@ -97,12 +104,18 @@ EtherPauseSource::cleanup(CleanupStage)
...
 EtherPauseSource::run_timer(Timer *)
 {
-    if (Packet *p = _packet->clone()) {
-	++_count;
-	output(0).push(p);
+    Packet *p = _packet->clone();
+    if (p) {
+        ++_count;
+    #if HAVE_BATCH
+        output_push_batch(0, PacketBatch::make_from_packet(p));
```
Comment (tbarbette) [inline elements/ethernet/etherpausesource.cc]: Same here

Comment (gkatsikas) [reply elements/ethernet/etherpausesource.cc]: Sure I can replace:
#if HAVE_BATCH
with
if (in_batch_mode == BATCH_MODE_YES) {

but I believe it is not necessary in this case.
Both elements are source elements (i.e., they have on input ports) and both of them have:
#if HAVE_BATCH
    in_batch_mode = BATCH_MODE_YES;
#endif

in their constructor, so the path begins from these elements and in_batch_mode is always BATCH_MODE_YES.

If you still believe I am wrong, I can push the changes :)

Comment (tbarbette) [reply elements/ethernet/etherpausesource.cc]: You are correct. I forgot they were source and therefore always in batch mode.

**Pull request merged**